### PR TITLE
Create get-proton-ge.sh

### PR DESCRIPTION
**Pull request by mdeguzis**
Add simple script to get the latest release tarball and install it. Tested on SteamOS / Debian.

### CODE REVIEW

File: bin/get-proton-ge.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		steamos=0
+		;;
+	*)
+		echo "ERROR: Unsupported type: \"${steam_type}\""
+		exit 1	
+		;;
+esac
+
+GIT_REPO="GloriousEggroll/proton-ge-custom"
+GIT_URL="https://github.com/GloriousEggroll/proton-ge-custom"
```
Comment (mmequignon) [inline bin/get-proton-ge.sh:31]:
```suggestion
GIT_URL="https://github.com/${GIT_REPO}"
```

File: bin/get-proton-ge.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+fi
+
+# Prepare dir(s)
+# Avoid any symlinked path jankiness
+# Do not quote path (issues on some systems such as SteamOS)
+if [[ ${flatpak} -eq 0 ]]; then
+	target=$(readlink -f ~/.var/app/com.valvesoftware.Steam/data/Steam/compatibilitytools.d)
+else
+	target=$(readlink -f ~/.steam/root/compatibilitytools.d)
+fi
```
Comment (mmequignon) [inline bin/get-proton-ge.sh:52]: You should ensure folder exists.

Comment (mdeguzis) [reply bin/get-proton-ge.sh]: Updated with path checks for steam root / compat folder.

**Comment by mmequignon** [comment]
I would have done this in Python, to avoid the grep/awk and so on.
Here is what I've done:
```python
#!/usr/bin/env python3
# Author Michael DeGuzis
# Description: Simple script to grab the latest Proton GE without a git clone / build.

import argparse
import json
import os
import tarfile
import shutil
import sys
from urllib import parse, request

PARSER = argparse.ArgumentParser(
    prog="proton-ge-downloader",
    description="Downloads the latest revision of GloriousEggroll's proton-ge."
)
PARSER.add_argument(
    "steam_type",
    choices=["flatpak", "native", "steamos"],
    help="Steam type. One of flatpak, native or steamos"
)

def get_response(url):
    return request.urlopen(url)

def get_verion(url):
    """Gets version name for the latest rev from github."""
    response = request.urlopen(url)
    data = response.read()
    content = json.loads(data)
    return content.get("tag_name")

GITHUB_URL = "https://github.com"
API_GITHUB_URL = "https://api.github.com"
GIT_REPO = "GloriousEggroll/proton-ge-custom/"
GITHUB_REPO_URL = parse.urljoin(GITHUB_URL, GIT_REPO)
RELEASE_URL = parse.urljoin(
    API_GITHUB_URL, "/".join(["repos", GIT_REPO, "releases/latest"])
)
VERSION = get_verion(RELEASE_URL)
FILENAME = "Proton-{}.tar.gz".format(VERSION)
FOLDER_NAME = "Proton-{}".format(VERSION)
DOWNLOAD_URL = parse.urljoin(
    GITHUB_REPO_URL, "/".join(["releases/download", VERSION, FILENAME])
)
USER = os.environ.get("USER")


def check_user(steam_type):
    """Ensure proper user if running SteamOS."""
    if steam_type == "steamos" and USER != "steam":
        print("ERROR: Please run this script as user 'steam'")
        sys.exit(1)

def get_target_path(steam_type):
    """Get path where files should be downloaded."""
    home = os.environ.get("HOME")
    if steam_type == "flatpak":
        path = os.path.join(
            home, ".var/app/com.valvesoftware.Steam/data/Steam"
        )
    else:
        path = os.path.join(home, ".steam/root")
    if not os.path.exists(path):
        print(
            "ERROR: Steam root does not exist! "
            "Expected path: {}".format(path)
        )
        sys.exit(1)
    compat_path = os.path.join(path, "compatibilitytools.d")
    if not os.path.exists(compat_path):
        print("WARNING: compatibilitytools.d folder mising, creating...")
        try:
            os.mkdir(compat_path)
        except IOError:
            print(
                "ERROR: Could not create compatibilitytools.d "
                "in Steam root: {}".format(path)
            )
            sys.exit(1)
    return compat_path

def download_file(url, target):
    """Downloads and extracts latest release to compatibilitytools.d."""
    out_folder = os.path.join(target, FOLDER_NAME)
    if os.path.exists(out_folder):
        print(
            "{} already exists in target_abs_path {}!".format(
                FOLDER_NAME, target
            )
        )
        response = input("Reset installation ? (y/N): ")
        if response not in ("y", "Y"):
            sys.exit(1)
        shutil.rmtree(out_folder)
    print("Downloading {}".format(FILENAME))
    response = request.urlopen(url)
    print("Extracting Proton GE to target_abs_path: {}".format(target))
    tar_file = tarfile.open(fileobj=response, mode="r|gz")
    tar_file.extractall(target)
    tar_file.close()
    print("{} Installed to {}!".format(FOLDER_NAME, target))

if __name__ == "__main__":
    args = PARSER.parse_args()
    check_user(args.steam_type)
    target_abs_path = get_target_path(args.steam_type)
    download_file(DOWNLOAD_URL, target_abs_path)
```
This is basically what you've done with bash.

**Comment by GloriousEggroll** [comment]
Hi,
While I do appreciate the tremendous amount of effort put into this, unfortunately it would end up being another addition that needs to be regularly maintained. Additionally I feel it's important for users to go through the manual installation process so that they know where to look to make modifications/changes. I also feel it should be independent from the main project because it is redundant to visit the repo to download the script when the entire release is on the same page.

**Pull request closed without merging**